waku.md

```diff
@@ -209,17 +209,29 @@ It is only relevant if you want to decrypt the incoming message, but if you only
 
 Data field contains encrypted message of the Envelope. In case of symmetric encryption, it also contains appended Salt (a.k.a. AES Nonce, 12 bytes). Plaintext (unencrypted) payload consists of the following concatenated fields: flags, auxiliary field, payload, padding and signature (in this sequence).
 
-	flags: 1 byte; first two bits contain the size of auxiliary field, third bit indicates whether the signature is present.
-	
-	auxiliary field: up to 4 bytes; contains the size of payload.
+Using [Augmented Backus-Naur form (ABNF)](https://tools.ietf.org/html/rfc5234) we have the following format:
+
+```
+; 1 byte; first two bits contain the size of auxiliary field, third bit indicates whether the signature is present.
+flags           = 1*OCTET
```

Comment:
I kinda like having bytes aliased AS OCTET, because it is more general and readable, but direct to OCTET seems fine too

Reply:
we can adjust later if we prefer that style